AutoCollection/Statsbeat.ts

```diff
@@ -209,42 +201,64 @@ class Statsbeat {
-        var elapsedMs = requests.time - lastRequests.time;
-        var averageRequestExecutionTime = ((this._intervalRequestExecutionTime - this._lastIntervalRequestExecutionTime) / intervalRequests) || 0; // default to 0 in case no requests in this interval
-        this._lastIntervalRequestExecutionTime = this._intervalRequestExecutionTime; // reset
-        if (elapsedMs > 0 && intervalRequests > 0) {
-            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_DURATION, value: averageRequestExecutionTime, properties: properties });
+    private _getNetworkStatsbeatCounter(category: number, endpoint: string): Network.NetworkStatsbeat {
+        // Check if counter is available
+        for (let i = 0; i < this._networkStatsbeatCollection.length; i++) {
+            // Same object
+            if (category === this._networkStatsbeatCollection[i].category &&
+                endpoint === this._networkStatsbeatCollection[i].endpoint) {
+                return this._networkStatsbeatCollection[i];
+            }
         }
-        this._lastRequests = requests;
+        // Create a new one if not found
+        let newCounter = new Network.NetworkStatsbeat(category, endpoint);
+        this._networkStatsbeatCollection.push(newCounter);
+        return newCounter;
     }
 
-    private _trackRequestsCount(properties: {}) {
-        if (this._totalSuccesfulRequestCount > 0) {
-            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_SUCCESS, value: this._totalSuccesfulRequestCount, properties: properties });
-            this._totalSuccesfulRequestCount = 0; //Reset
-        }
-        if (this._totalFailedRequestCount > 0) {
-            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_FAILURE, value: this._totalFailedRequestCount, properties: properties });
-            this._totalFailedRequestCount = 0; //Reset
-        }
-        if (this._retryCount > 0) {
-            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.RETRY_COUNT, value: this._retryCount, properties: properties });
-            this._retryCount = 0; //Reset
-        }
-        if (this._throttleCount > 0) {
-            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.THROTTLE_COUNT, value: this._throttleCount, properties: properties });
-            this._throttleCount = 0; //Reset
+    private _trackRequestDuration(commonProperties: {}) {
+        for (let i = 0; i < this._networkStatsbeatCollection.length; i++) {
+            var currentCounter = this._networkStatsbeatCollection[i];
+            currentCounter.time = +new Date;
+            var intervalRequests = (currentCounter.totalRequestCount - currentCounter.lastRequestCount) || 0;
+            var elapsedMs = currentCounter.time - currentCounter.lastTime;
+            var averageRequestExecutionTime = ((currentCounter.intervalRequestExecutionTime - currentCounter.lastIntervalRequestExecutionTime) / intervalRequests) || 0;
```

Comment:
Could there be a divide by zero error thrown here?

Reply:
Node.js doesn't throw an error with division by zero, it will return "Infinity" and metric will not be send because intervalRequests would be 0